{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "220b879c",
   "metadata": {},
   "outputs": [],
   "source": [
    "from imutils import face_utils\n",
    "import numpy as np\n",
    "import imutils\n",
    "import dlib\n",
    "import cv2\n",
    "import os\n",
    "from matplotlib import pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65216fcc",
   "metadata": {},
   "outputs": [],
   "source": [
    "!pip install mlxtend"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "615193c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import math\n",
    "import matplotlib.pyplot as plt\n",
    "from mlxtend.image import extract_face_landmarks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "de4d8255",
   "metadata": {},
   "outputs": [],
   "source": [
    "datFile =  \"shape_predictor_68_face_landmarks.dat\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0edf6657",
   "metadata": {},
   "outputs": [],
   "source": [
    "def downsampling(img):\n",
    "    g_down = []\n",
    "    g_down.append(img)\n",
    "\n",
    "    for i in range(3):\n",
    "        tmp = cv2.pyrDown(img)\n",
    "        g_down.append(tmp)\n",
    "        img = tmp\n",
    "\n",
    "    cv2.waitKey(0)\n",
    "    cv2.destroyAllWindows()\n",
    "\n",
    "    return g_down[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "dc64aaae",
   "metadata": {},
   "outputs": [],
   "source": [
    "def show_raw_detection(image, detector, predictor):\n",
    "    gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)\n",
    "    rects = detector(image, 1)\n",
    "\n",
    "    for (i, rect) in enumerate(rects):\n",
    "        shape = predictor(gray, rect)\n",
    "        shape = face_utils.shape_to_np(shape)\n",
    "\n",
    "        (x,y,w,h) = face_utils.rect_to_bb(rect)\n",
    "        cv2.rectangle(image, (x,y), (x+w,y+h),(0,255,0),2)\n",
    "\n",
    "        cv2.putText(image, \"Face #{}\".format(i+1), (x-10,y-10), cv2.FONT_HERSHEY_SIMPLEX, 0.5, (0,255,0), 2)\n",
    "\n",
    "        for (x,y) in shape:\n",
    "            cv2.circle(image, (x,y), 1, (0,0,255), -1)\n",
    "\n",
    "    cv2.imshow(\"\",image)\n",
    "    cv2.waitKey(0)\n",
    "    cv2.destroyAllWindows()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "e511308a",
   "metadata": {},
   "outputs": [],
   "source": [
    "from PIL import Image\n",
    "\n",
    "def read_images_from_folder(folder_path):\n",
    "    image_list = []\n",
    "\n",
    "    for filename in os.listdir(folder_path):\n",
    "        # 이미지 파일인지 확인\n",
    "        if filename.endswith(\".jpg\") or filename.endswith(\".png\") or filename.endswith(\".jpeg\"):\n",
    "            image_path = os.path.join(folder_path, filename)\n",
    "            try:\n",
    "                # 이미지 파일 열기\n",
    "                img = Image.open(image_path)\n",
    "                # 이미지 리스트에 추가\n",
    "                image_list.append(img)\n",
    "            except Exception as e:\n",
    "                # 이미지 파일 열기에 실패한 경우 예외 처리\n",
    "                print(f\"Failed to open image: {image_path}\\nError: {str(e)}\")\n",
    "\n",
    "    return image_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "bedbb9a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 이미지가 있는 폴더 경로 설정\n",
    "folder_path = \"sad/\"\n",
    "\n",
    "# 폴더 내 이미지 읽어오기\n",
    "images = read_images_from_folder(folder_path)\n",
    "# print(images)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c0912c38",
   "metadata": {},
   "outputs": [],
   "source": [
    "import uuid\n",
    "import warnings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0f9ef155",
   "metadata": {},
   "outputs": [],
   "source": [
    "def generate_random_filename(extension):\n",
    "    random_name = uuid.uuid4().hex\n",
    "    filename = f\"{random_name}.{extension}\"\n",
    "    return filename"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b58bae53",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\anaconda3\\lib\\site-packages\\mlxtend\\image\\extract_face_landmarks.py:67: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n",
      "C:\\Users\\ballj\\AppData\\Local\\Temp\\ipykernel_2280\\3684999397.py:13: UserWarning: No face detected.\n",
      "  warnings.warn(\"No face detected.\")\n"
     ]
    },
    {
     "ename": "MemoryError",
     "evalue": "Unable to allocate 20.5 MiB for an array with shape (2316, 3088, 3) and data type uint8",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mMemoryError\u001b[0m                               Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[10], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m image \u001b[38;5;129;01min\u001b[39;00m images:\n\u001b[1;32m----> 2\u001b[0m     image_np \u001b[38;5;241m=\u001b[39m cv2\u001b[38;5;241m.\u001b[39mcvtColor(\u001b[43mnp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43marray\u001b[49m\u001b[43m(\u001b[49m\u001b[43mimage\u001b[49m\u001b[43m)\u001b[49m, cv2\u001b[38;5;241m.\u001b[39mCOLOR_RGB2BGR)\n\u001b[0;32m      3\u001b[0m     downsample_face \u001b[38;5;241m=\u001b[39m downsampling(image_np)\n\u001b[0;32m      5\u001b[0m     detector \u001b[38;5;241m=\u001b[39m dlib\u001b[38;5;241m.\u001b[39mget_frontal_face_detector()\n",
      "\u001b[1;31mMemoryError\u001b[0m: Unable to allocate 20.5 MiB for an array with shape (2316, 3088, 3) and data type uint8"
     ]
    }
   ],
   "source": [
    "for image in images:\n",
    "    image_np = cv2.cvtColor(np.array(image), cv2.COLOR_RGB2BGR)\n",
    "    downsample_face = downsampling(image_np)\n",
    "\n",
    "    detector = dlib.get_frontal_face_detector()\n",
    "    predictor = dlib.shape_predictor(datFile)\n",
    "    image = imutils.resize(downsample_face, width=500)\n",
    "#     show_raw_detection(image, detector, predictor)\n",
    "\n",
    "    landmarks = extract_face_landmarks(image)\n",
    "    \n",
    "    if not detector(image, 1):\n",
    "        warnings.warn(\"No face detected.\")\n",
    "        None\n",
    "    else:\n",
    "        fig = plt.figure(figsize=(15, 5))\n",
    "        ax = fig.add_subplot(1, 3, 2)\n",
    "        ax.scatter(landmarks[:, 0], -landmarks[:, 1], alpha=0.8)\n",
    "    \n",
    "        # 결과 이미지 파일 경로\n",
    "        output_folder = \"sad_new/\"\n",
    "\n",
    "        output_file = os.path.join(output_folder,generate_random_filename(\"jpg\"))\n",
    "\n",
    "        fig.savefig(output_file)\n",
    "        plt.close(fig)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0dc4aecb",
   "metadata": {},
   "outputs": [],
   "source": [
    "def crop_images_in_folder(input_folder, output_folder, x, y,w,h):\n",
    "    # 출력 폴더가 없는 경우 생성\n",
    "    if not os.path.exists(output_folder):\n",
    "        os.makedirs(output_folder)\n",
    "\n",
    "    # 입력 폴더의 이미지 파일들을 처리\n",
    "    for filename in os.listdir(input_folder):\n",
    "        # 이미지 파일인지 확인\n",
    "        if filename.endswith(\".jpg\") or filename.endswith(\".png\") or filename.endswith(\".jpeg\"):\n",
    "            image_path = os.path.join(input_folder, filename)\n",
    "\n",
    "            # 이미지 불러오기\n",
    "            image = cv2.imread(image_path)\n",
    "\n",
    "            # 이미지가 정상적으로 불러와졌을 경우에만 처리\n",
    "            if image is not None:\n",
    "                # 이미지 자르기\n",
    "                cropped_image = image[y:y+h, x:x+w]\n",
    "\n",
    "                # 저장할 파일 경로 생성\n",
    "                output_path = os.path.join(output_folder, filename)\n",
    "\n",
    "                # 이미지 저장\n",
    "                cv2.imwrite(output_path, cropped_image)\n",
    "\n",
    "#                 print(f\"Cropped image saved: {output_path}\")\n",
    "\n",
    "# 입력 폴더와 출력 폴더 경로 설정\n",
    "input_folder = \"sad_new/\"\n",
    "output_folder = \"sad_final/\"\n",
    "\n",
    "# 이미지 자르기 및 저장\n",
    "x,y = 590, 50\n",
    "w,h = 370, 400\n",
    "crop_images_in_folder(input_folder, output_folder, x, y,w,h)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "90cc90b7",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
